src/utils/functions.ts

```diff
@@ -9,4 +14,30 @@ export const promiseConcurrent = <T, R>(
 
 export const numberWithCommas = (x: number | bigint | string) => {
   return x.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",");
-}
\ No newline at end of file
+}
+
+export async function waitTxDone(api: ApiPromise, tx: SubmittableExtrinsic, sender: KeyringPair, options?: Partial<SignerOptions>): Promise<string> {
```

Comment:
This should be part of the transactions.ts

Reply:
Also it might already exist there.

Reply:
I noticed the `sendAllStreamAndWaitLast` function however, it wasn't  giving the necessary events for debugging, as specified here: https://polkadot.js.org/docs/api/cookbook/tx/#how-do-i-get-the-decoded-enum-for-an-extrinsicfailed-event
It also seemed quite specific for a pre-defined use case. 

Does it make sense to align both these functions to be one?

Reply:
If it is not disruptive yes, otherwise you can keep separate for now

Reply:
I gave it a shot here https://github.com/PureStake/moonbeam-tools/pull/8

did not test it locally though.